pkg/p2p/key.go

```diff
@@ -0,0 +1,45 @@
+package p2p
+
+import (
+	"io/ioutil"
+
+	"github.com/tendermint/tendermint/internal/p2p"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	tmos "github.com/tendermint/tendermint/libs/os"
+	"github.com/tendermint/tendermint/types"
+)
+
+// LoadNodeKey loads NodeKey located in filePath.
+func LoadNodeKey(filePath string) (p2p.NodeKey, error) {
```

Comment:
This has a dependency on an internal type (`p2p.NodeKey`) which is internal (and therefore difficult to use by callers outside of this package, and is (confusingly) sort of the same as having a function that has `func Exported() unexported {}`, which is allowable by the compiler but not super useful (and I think shouldn't be allowed, but here we are).

Reply:
Good catch.

Reply:
now it works only with nodeID which is public